### PR TITLE
Boxgrps to spangrps tool

### PR DESCRIPTION
**Pull request by geli-gel**
Fix for https://github.com/allenai/scholar/issues/36624
tool to get spangroups from boxgroups using new `is_overlap(center=True)` logic to emulate LayoutParser's .filter_by(center=True) which was being used in bib entry detector local testing, and giving better texts than SPP which was using default MMDA `_annotate_box_groups`. 

Basically takes `_annotate_box_groups` and moves it to `tools.py` and adds a couple changes (like default to assume we want centers of tokens only, and adds a little `x` padding to make sure we grab those narrow tokens like "[" which would be missed without the extra padding.

### CODE REVIEW

**Review by cmwilhelm — CHANGES_REQUESTED** [review]
This was fast!

Angele, it looks like you've factored the code out of `document._annotate_box_groups()`, but I don't see any changes to `document.py` itself.

We should probably update `Document._annotate_box_groups` to call through to the utils fn with the correct arguments.

I also made a comment about verifying that this didn't change existing behavior, which might be easier to do after the Document changes.

File: src/mmda/utils/tools.py [hunk 1]
```diff
@@ -1,36 +1,154 @@
...
+            else:
+                cur_page_tokens = all_page_tokens[box.page]
+
+            # Find all the tokens within the box
+            tokens_in_box, remaining_tokens = allocate_overlapping_tokens_for_box(
+                tokens=cur_page_tokens,
+                box=box,
+                token_box_in_box_group=token_box_in_box_group,
+                # pad a small amount so that extra narrow token boxes (when split at punctuation) are not missed
+                x=avg_token_widths.get(box.page, 0.0) * 0.5,
```
Comment (cmwilhelm) [inline src/mmda/utils/tools.py:99]: If we want to maintain OG behavior for center=False, we should not pad in that case. Or make padding an option to this fn.

File: src/mmda/utils/tools.py [hunk 2]
```diff
@@ -1,36 +1,154 @@
...
     return allocated_tokens, remaining_tokens
 
 
+def box_groups_to_span_groups(
+        box_groups: List[BoxGroup], doc: Document, center: bool = True
+) -> List[SpanGroup]:
+    """Generate SpanGroups from BoxGroups.
+    Args
+        `box_groups` (List[BoxGroup])
+        `doc` (Document) base document annotated with pages, tokens, rows to
```
[the remark this answers is not in the document]
Comment (geli-gel) [reply src/mmda/utils/tools.py:49]: oh, yeah probably just needs pages and tokens! I can remove from the fixture and see how it goes

[the remark this answers is not in the document]
Comment (cmwilhelm) [reply src/mmda/utils/tools.py:49]: What was the outcome of this?

File: tests/test_types/test_document.py [hunk 3]
```diff
@@ -73,3 +74,34 @@ def test_annotate_box_groups_gets_text(self):
 
         doc.annotate(new_span_groups=box_groups)
         assert doc.new_span_groups[0].text.startswith("Gutman G, Rosenzweig D, Golan J")
+
+    def test_annotate_box_groups_allocates_all_overlapping_tokens(self):
```
Comment (cmwilhelm) [inline tests/test_types/test_document.py:78]: Might be worth adding a docstring to this test to indicate it's a regression test to verify annotating box groups onto a doc produces the same output as a last-known-good fixture

**Review by cmwilhelm — APPROVED** [review]
LGTM!

**Pull request merged**